src/main/scala/com/snowflake/snowpark/functions.scala

```diff
@@ -3140,6 +3140,112 @@ object functions {
    */
   def listagg(col: Column): Column = listagg(col, "", isDistinct = false)
 
+  /**
+   * This function receives a column and extracts the groupIdx from the string
+   * after applying the exp regex. Returns empty string when the string doesn't
+   *  match and null if the input is null.
+   *
+   * This function applies the `case sensitive` and `extract` flags.
+   * It doesn't apply multiline nor .* matches newlines.
+   * If these flags need to be applied, use `builtin("REGEXP_SUBSTR")`
+   * instead and apply the desired flags.
+   *
+   * Note: non-greedy tokens such as `.*?` are not supported
+   * @since 1.12.1
+   * @param colName Column to apply regex.
+   * @param exp Regex expression to apply.
+   * @param grpIdx Group to extract.
+   * @return Column object.
+   */
+  def regexp_extract(
+      colName: Column,
+      exp: String,
+      position: Int,
+      Occurences: Int,
+      grpIdx: Int): Column = {
+    when(colName.is_null, lit(null))
+      .otherwise(
+        coalesce(
+          builtin("REGEXP_SUBSTR")(
+            colName,
+            lit(exp),
+            lit(position),
+            lit(Occurences),
+            lit("ce"),
+            lit(grpIdx)),
+          lit("")))
+  }
+
+  /**
+   * Returns the sign of the given column. Returns either 1 for positive,
+   * 0 for 0 or
+   * NaN, -1 for negative and null for null.
+   * NOTE: if string values are provided snowflake will attempts to cast.
+   *  If it casts correctly, returns the calculation,
+   *  if not an error will be thrown
+   * @since 1.12.1
+   * @param e Column to calculate the sign.
+   * @return Column object.
+   */
+  def signum(colName: Column): Column = {
+    builtin("SIGN")(colName)
+  }
+
+  /**
+   * Returns the sign of the given column. Returns either 1 for positive,
```

Comment:
align with: https://github.com/snowflakedb/snowpark-python/blob/7e055720096ad91ccf2733abbb47a00afa5ea6cb/src/snowflake/snowpark/functions.py#L2514 provide examples

Reply:
Modified description and added examples.Could you please review.